src/server/database/leveldb-adaptor.js

```diff
@@ -0,0 +1,86 @@
+'use strict'
+
+const levelup = require('levelup')
+const PapanUtils = require('../utils.js')
+
+class LevelCollectionAdaptor {
+  constructor (db) {
+    this.db = db
+  }
+
+  put (doc) {
+    let db = this.db
+
+    return new Promise((resolve, reject) => {
+      db.put(doc._id, JSON.stringify(doc), (err) => {
+        if (err) {
+          reject(err)
+        } else {
+          resolve()
+        }
+      })
+    })
+  }
+
+  get (key) {
+    let db = this.db
+
+    return new Promise((resolve, reject) => {
+      db.get(key, (err, value) => {
+        if (err) {
+          reject(err)
+        } else {
+          resolve(JSON.parse(value))
+        }
+      })
+    })
+  }
+
+  update (doc) {
+    return this.put(doc)
+  }
+
+  remove (doc) {
+    let db = this.db
+
+    return new Promise((resolve, reject) => {
+      db.del(doc._id, (err) => {
+        if (err) {
+          reject(err)
+        } else {
+          resolve()
+        }
+      })
+    })
+  }
+}
+
+class LevelAdaptor {
+  table (name) {
+    return new Promise((resolve, reject) => {
+      let fullURL = this.url + name + '.json'
+      let db = levelup(fullURL, { db: require('jsondown') })
+      resolve(new LevelCollectionAdaptor(db))
+    })
+  }
+
+  connect (url = 'database') {
+    if (!url.endsWith('/') && url.length !== 0) {
+      this.url = url + '/'
```

Comment:
If this.url is a property and reused in other methods, make declaration maybe more obvious? For instance in a constructor with default value 'database'?